src/groundlight/binary_labels.py

```diff
@@ -1,51 +1,98 @@
-"""Defines the possible values for binary class labels like Yes/No or PASS/FAIL.
+"""Defines the possible values for binary class labels like YES/NO.
 Provides methods to convert between them.
 
 This part of the API is kinda ugly right now.  So we'll encapsulate the ugliness in one place.
 """
 import logging
-from typing import List, Union
+from enum import Enum
+from typing import Any, Union
 
-from model import Detector, ImageQuery
+from model import ClassificationResult, Detector, ImageQuery
 
-logger = logging.getLogger("groundlight.sdk")
+logger = logging.getLogger(__name__)
 
 
-def internal_labels_for_detector(
-    context: Union[ImageQuery, Detector, str],  # pylint: disable=unused-argument
-) -> List[str]:
-    """Returns an ordered list of class labels as strings.
-    These are the versions of labels that the API demands.
-    :param context: Can be an ImageQuery, a Detector, or a string-id for one of them.
-    """
-    # NOTE: At some point this will need to be an API call, because these will be defined per-detector
-    return ["PASS", "FAIL"]
+class Label(str, Enum):
+    YES = "YES"
+    NO = "NO"
+    UNSURE = "UNSURE"
+
+
+VALID_DISPLAY_LABELS = {Label.YES, Label.NO, Label.UNSURE}
+
+
+class DeprecatedLabel(str, Enum):
+    PASS = "PASS"
+    FAIL = "FAIL"
+    NEEDS_REVIEW = "NEEDS_REVIEW"
+
+
+DEPRECATED_LABEL_NAMES = {DeprecatedLabel.PASS, DeprecatedLabel.FAIL, DeprecatedLabel.NEEDS_REVIEW}
 
 
 def convert_internal_label_to_display(
     context: Union[ImageQuery, Detector, str],  # pylint: disable=unused-argument
     label: str,
 ) -> str:
+    """Convert a label that comes from our API into the label string that we show to the user.
+
+    NOTE: We return UPPERCASE label strings to the user, unless there is a custom label (which
+    shouldn't be happening at this time).
+    """
     # NOTE: Someday we will do nothing here, when the server provides properly named classes.
+    if not isinstance(label, str):
+        raise ValueError(f"Expected a string label, but got {label} of type {type(label)}")
     upper = label.upper()
-    if upper == "PASS":
-        return "YES"
-    if upper == "FAIL":
-        return "NO"
-    if upper in ["YES", "NO"]:
-        return label
+    if upper in {Label.YES, DeprecatedLabel.PASS}:
+        return Label.YES.value
+    if upper in {Label.NO, DeprecatedLabel.FAIL}:
+        return Label.NO.value
+    if upper in {Label.UNSURE, DeprecatedLabel.NEEDS_REVIEW}:
+        return Label.UNSURE.value
+
+    # Safety versus extensibility: should it be an error if we get an unrecognized internal label?
     logger.warning(f"Unrecognized internal label {label} - leaving alone.")
```

Comment:
In theory, this shouldn't happen because we don't support custom labels. But if it does (e.g., we push a change to the API to support custom labels, or we return a weird label because of some internal error), do we want to have to explicitly turn on support here too?